test/output-app/model/Tweet.js

```diff
@@ -1,60 +1,115 @@
+import log from '../server/logger';
 import DataLoader from 'dataloader';
-import findByIds from 'mongo-find-by-ids';
+import { findByIds, queryForRoles, authlog, checkAuthDoc } from '../server/authorize';
 
 export default class Tweet {
   constructor(context) {
     this.context = context;
     this.collection = context.db.collection('tweet');
     this.pubsub = context.pubsub;
-    this.loader = new DataLoader(ids => findByIds(this.collection, ids));
+    let authQuery;
+    try {
+      const { me, User } = context;
+      authQuery = queryForRoles(me, ['admin', 'world'], ['authorId', 'coauthorsIds'], { User }, authlog('tweet findOneById', 'readOne', me));
+    } catch (err) { 
+      log.error(err.message);
+      authQuery = {_id: false}; // otherwise admin access
+    }
+    this.authorizedLoader = new DataLoader(ids => findByIds(this.collection, ids, authQuery));
   }
 
-  findOneById(id) {
-    return this.loader.load(id);
+  async findOneById(id, me, resolver) {
+    try {
+      return await this.authorizedLoader.load(id);
+    } catch (err) { log.error(err.message); }
   }
 
-  all({ lastCreatedAt = 0, limit = 10 }) {
-    return this.collection.find({
-      createdAt: { $gt: lastCreatedAt },
-    }).sort({ createdAt: 1 }).limit(limit).toArray();
+  find({ lastCreatedAt = 0, limit = 10, baseQuery = { createdAt: { $gt: lastCreatedAt } } }, me, resolver) {
+    try {
+      const authQuery = queryForRoles(me, ['admin', 'world'], ['authorId', 'coauthorsIds'], { User: this.context.User }, authlog(resolver, 'readMany', me));
+      const finalQuery = {...baseQuery, ...authQuery};
+      return this.collection.find(finalQuery).sort({ createdAt: 1 }).limit(limit).toArray();
+    } catch (err){ log.error(err.message); }
```

Comment:
Removing it in model/Twitter.js currently has no negative effect on the current test cases.

But if I remove try/catch in the model/User.js, it is getting admin access in some of the cases. Or maybe we have to adjust the underlying queryForRoles -> logger(authlog) instead, which holds the throw on error? Do you have a better idea therefore? I agree the try/catch here isn't nice, but somehow I have to react on this error case, otherwise I'll get admin access.